File: twopointer/containermaxwater/ContainerWithMostWater.java
package containermaxwater;

public class ContainerWithMostWater {

    public int maxArea(int[] height) {
        int left = 0;  // Left pointer
        int right = height.length - 1;  // Right pointer
        int maxWater = 0;  // Store the maximum water area found

        while (left < right) {
            // Calculate the area between the current pair of lines
            int width = right - left;
            int currentHeight = Math.min(height[left], height[right]);
            int currentArea = width * currentHeight;

            // Update the maximum water area if the current area is larger
            maxWater = Math.max(maxWater, currentArea);

            // Move the pointer corresponding to the shorter line
            if (height[left] < height[right]) {
                left++;  // Move the left pointer inward
            } else {
                right--;  // Move the right pointer inward
            }
        }

        return maxWater;  // Return the maximum area found
    }

    public static void main(String[] args) {
        ContainerWithMostWater solution = new ContainerWithMostWater();

        // Test cases
        int[] height1 = {1, 8, 6, 2, 5, 4, 8, 3, 7};
        int[] height2 = {1, 1};

        System.out.println("Max Water Area (Test 1): " + solution.maxArea(height1));  // Output: 49
        System.out.println("Max Water Area (Test 2): " + solution.maxArea(height2));  // Output: 1
    }
}

File: twopointer/containermaxwater/ContainerWithMostWater.py
class ContainerWithMostWater:
    def max_area(self, height):
        left = 0  # Left pointer
        right = len(height) - 1  # Right pointer
        max_water = 0  # To store the maximum water area

        while left < right:
            # Calculate the area between the current pair of lines
            width = right - left
            current_height = min(height[left], height[right])
            current_area = width * current_height

            # Update max_water if the current area is larger
            max_water = max(max_water, current_area)

            # Move the pointer corresponding to the shorter line inward
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return max_water  # Return the maximum water area

# Test cases
if __name__ == "__main__":
    solution = ContainerWithMostWater()

    # Test cases
    height1 = [1, 8, 6, 2, 5, 4, 8, 3, 7]
    height2 = [1, 1]

    print("Max Water Area (Test 1):", solution.max_area(height1))  # Output: 49
    print("Max Water Area (Test 2):", solution.max_area(height2))  # Output: 1
